addons/middle-click-popup/userscript.js

```diff
@@ -0,0 +1,470 @@
+
+      let count = 0;
+
+      let split = val.split(" ");
+      let listLI = this.dropdown.getElementsByTagName("li");
+      for (const li of listLI) {
+        const procCode = li.data.text;
+        const lower = li.data.lower;
+        // let i = li.data.lower.indexOf(val);
+        // let array = regExp.exec(li.data.lower);
+
+        let im = 0;
+        let match = [];
+        for (let si = 0; si < split.length; si++) {
+          let find = " " + split[si];
+          let idx = lower.indexOf(find, im);
+          if (idx === -1) {
+            match = null;
+            break;
+          }
+          match.push(idx);
+          im = idx + find.length;
+        }
+
+        if (count < this.DROPDOWN_BLOCK_LIST_MAX_ROWS && match) {
+          li.style.display = "block";
+          while (li.firstChild) {
+            li.removeChild(li.firstChild);
+          }
+
+          let i = 0;
+
+          for (let iM = 0; iM < match.length; iM++) {
+            let im = match[iM];
+            if (im > i) {
+              li.appendChild(document.createTextNode(procCode.substring(i, im)));
+              i = im;
+            }
+            let bText = document.createElement("b");
+            let len = split[iM].length;
+            bText.appendChild(document.createTextNode(procCode.substr(i, len)));
+            li.appendChild(bText);
+            i += len;
+          }
+
+          if (i < procCode.length) {
+            li.appendChild(document.createTextNode(procCode.substr(i)));
+          }
+
+          if (count === 0) {
+            li.classList.add("sel");
+          } else {
+            li.classList.remove("sel");
+          }
+          count++;
+        } else {
+          li.style.display = "none";
+          li.classList.remove("sel");
+        }
+      }
+      p.append(this.dropdown);
+    }
+
+    inputKeyDown(e) {
+      if (e.keyCode === 38) {
+        this.navigateFloatFilter(-1);
+        e.preventDefault();
+        return;
+      }
+      if (e.keyCode === 40) {
+        this.navigateFloatFilter(1);
+        e.preventDefault();
+        return;
+      }
+      if (e.keyCode === 13) {
+        // Enter
+        let sel = this.dropdown.querySelector(".sel");
+        if (sel) {
+          this.onClick(e);
+          this.hide();
+        }
+        e.cancelBubble = true;
+        e.preventDefault();
+        return;
+      }
+      if (e.keyCode === 27) {
+        // Escape
+        if (this.floatInput.value.length > 0) {
+          this.floatInput.value = ""; // Clear search first, then close on second press
+          this.inputChange(e);
+        } else {
+          this.hide();
+        }
+        e.preventDefault();
+        return;
+      }
+    }
+
+    buildFilterList() {
+      let options = [];
+
+      let t = this.workspace.getToolbox();
+
+      let blocks = t.flyout_.getWorkspace().getTopBlocks();
+      // 107 blocks, not in order... but we can sort by y value or description right :)
```

Comment:
Check if `t` is undefined and return if it is. This can happen if you press ctrl + space while in the custom block creation GUI, for example.